### PR TITLE
Update FSharp.Analyzers.SDK and related deps

### PR DESCRIPTION
**Pull request by Numpsy**
The existing version didn't seem to work for me in the latest Ionide, so - an attempt to update it to match the latest Ionide.Analyzers (a local debug build seems to work for me in Ionide 7.20 at least)

### CODE REVIEW

File: tests/FSharp.Analyzers.Tests/FSharp.Analyzers.Tests.fsproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
     <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
```
Comment (nojaf) [inline tests/FSharp.Analyzers.Tests/FSharp.Analyzers.Tests.fsproj:4]: Is this necessary?

File: .config/dotnet-tools.json [hunk 2]
```diff
@@ -6,19 +6,22 @@
       "version": "6.3.0-alpha-007",
       "commands": [
         "fantomas"
-      ]
+      ],
+      "rollForward": false
```
Comment (nojaf) [inline .config/dotnet-tools.json:10]: This looks odd to me. Is this a conscious change?

File: tests/FSharp.Analyzers.Tests/FSharp.Analyzers.Tests.fsproj (hunk 1 above)
Comment (Numpsy) [reply tests/FSharp.Analyzers.Tests/FSharp.Analyzers.Tests.fsproj:4]: FSharp.Analyzers.SDK.Testing appears to be .NET 7.0+ now, so I think it needs to be updated to 7 or 8 (just picked 8 as 7 is out of support now)

**Comment by nojaf** [comment]
Could you bump the SDK to 8.0.300 and the tools to their latest stable version?

**Comment by Numpsy** [comment]
> > The existing version didn't seem to work for me in the latest Ionide
> 
> Thanks, although I would like to point out that is literally hasn't been 24 hours since this mismatch 😅.

Yeah, just happened to cooincide with me trying to update some things at work :-)

File: tests/FSharp.Analyzers.Tests/FSharp.Analyzers.Tests.fsproj (hunk 1 above)
Comment (nojaf) [reply tests/FSharp.Analyzers.Tests/FSharp.Analyzers.Tests.fsproj:4]: Yep, sounds reasonable. I wasn't sure if that change got reverted or not.

**Comment by Numpsy** [comment]
Is it worth updating the github actions at the same time? (there are some build warnings about deprecated versions)
image

**Comment by nojaf** [comment]
Yep, that sounds good.

**Review by nojaf — DISMISSED** [review]
Thank you!

**Review by nojaf — APPROVED** [review]

**Review by marcin-krystianc — APPROVED** [review]

**Comment by nojaf** [comment]
https://www.nuget.org/packages/G-Research.FSharp.Analyzers/0.10.0 is available.

**Pull request merged**